### PR TITLE
docs: merged coding_style_standards.md with #820

### PR DESCRIPTION
**Pull request by Jaskowicz1**
This PR merges `coding_style_standards.md` with #820. This is because, previously, the coding standards were slightly incorrect and needed fixing (as requested by Brain).

- [x] My pull request is made against the `dev` branch.
- [x] I have ensured that the changed library can be built on your target system. I did not introduce any platform-specific code.
- [x] I have ensured that all methods and functions are **fully documented** using doxygen style comments.
- [x] I tested my commits, by adding a test case to the unit tests if needed
- [x] I have ensured that I did not break any existing API calls.
- [x] My code follows the [coding style guide](https://dpp.dev/coding-standards.html) (if you are not sure, match the code style of existing files including indent style etc).
- [x] I have not built my pull request using AI, a static analysis tool or similar without any human oversight. Where I have generated this pull request using a tool, I have justified why this is needed.

### CODE REVIEW

**Comment by Commandserver** [comment]
What's the purpose of this PR? the changes are already made in #820

**Comment by Jaskowicz1** [comment]
> What's the purpose of this PR? the changes are already made in #820

The current coding_style_standards have a slight issue with them (contradicting itself). Rather than just pushing a fix for that, I thought it would be ideal to just drag over the change in #820 into here as that already has the fix but expands upon a couple sections.

**Review by Commandserver — APPROVED** [review]

**Comment by Jaskowicz1** [comment]
Requested Review again from you @Commandserver as I've made another change, adding a new section. Don't want you to miss it in-case you don't like something about it.

**Review by Commandserver — APPROVED** [review]
Looks good :D

File: docpages/advanced_reference/coding_style_standards.md [hunk 1]
```diff
@@ -64,38 +92,47 @@ public:
...
 
-## C++ version
+Do not introduce platform specific (e.g. Windows only) code or libc functions. If you really must use these functions safely wrap them e.g. in `#ifdef _WIN32` and provide a cross-platform alternative so that it works for everyone.
 
-The code must work with the C++17 standard. 
+## C++ Version
 
-## Select the right size type for numeric types
-If a value will only hold values up to 255, use `uint8_t`. If a value cannot hold over 65536, use `uint16_t`. These types can help use a lot less ram at scale.
+The code must work with the C++17 standard.
```
Comment (Mishura4) [inline docpages/advanced_reference/coding_style_standards.md:125]: How about adding something like "unless for an optional feature that can be enabled and that uses a more recent standard, e.g. coroutines"?

Comment (Jaskowicz1) [reply docpages/advanced_reference/coding_style_standards.md:125]: Yep, I like that idea. I'll get onto it :)

**Pull request merged**